### PR TITLE
Support the ability to run e2e tests on different testbeds/NICs

### PR DESCRIPTION
**Pull request by akiselev1**
This PR will address #139
 
The proposed solution uses a new label (e2e-test-710) to initiate a e2e flow on 710-labelled runner. The original e2e-test  label is used to run test on a 810-labeled runner. Manual trigger (workflow_dispatch) allows to choose wither 810 (default) or 710 NIC hw to run.  All scenarios were tested in my fork: https://github.com/akiselev1/rhel-sriov-test/pull/1

### CODE REVIEW

**Review by dkosteck — APPROVED** [review]
Overall looks good, just my one comment above which is optional/something to consider

File: .github/workflows/e2e.yaml
[the remark this answers is not in the document]
Comment (akiselev1) [reply .github/workflows/e2e.yaml:1]: Good point. I was on the fence about this change. Will do.

**Comment by akiselev1** [comment]
@jianzzha - please take a look

File: .github/workflows/e2e810.yaml [hunk 1]
```diff
@@ -6,8 +6,8 @@ on:
   workflow_dispatch:
 jobs:
   e2e-tests:
-    runs-on: self-hosted
-    if: contains(github.event.label.name, 'e2e-test')
+    if: contains(github.event.label.name, 'e2e-test') || (github.event_name == 'workflow_dispatch')
+    runs-on: [self-hosted, 810]
```
Comment (jianzzha) [inline .github/workflows/e2e810.yaml:10]: Is [...] a OR or AND logic? If it is OR logic, Shouldn't this be: [810]?

**Comment by jianzzha** [comment]
I'm not sure I like the idea of duplicating the same code in different workflow files. If more NIC are added, then more duplication ... Instead, I think we should explore an idea of dynamic setting of runs-on

File: .github/workflows/e2e810.yaml (hunk 1 above)
Comment (akiselev1) [reply .github/workflows/e2e810.yaml:10]: It is AND logic. Omitting self-hosted label will work but it is not recommended in Github documentation.

**Comment by akiselev1** [comment]
> I'm not sure I like the idea of duplicating the same code in different workflow files. If more NIC are added, then more duplication ... Instead, I think we should explore an idea of dynamic setting of runs-on

I don't like it either. I spent a day experimenting with workarounds. As I mentioned in the PR description:

Unfortunately it is not possible to parameterize "runs-on" value for github action flow due to this known issue: https://github.com/actions/runner/issues/480 . Existing workarounds work only for a single label or require 2 files per flow.

File: .github/workflows/e2e810.yaml (hunk 1 above)
Comment (dkosteck) [reply .github/workflows/e2e810.yaml:10]: Just confirming what Andrew said
From: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idruns-on
"If you specify an array of strings, your workflow will execute on any runner that matches **all** of the specified runs-on values."

**Review by jianzzha — APPROVED** [review]
Looks ok. The last line of duplicate code is a trade off for less complexity and we can live with that

**Pull request merged**